sycl/source/detail/queue_impl.hpp

```diff
@@ -542,60 +540,14 @@ class queue_impl {
     }
     ur_result_t Error = Adapter->call_nocheck<UrApiKind::urQueueCreate>(
         Context, Device, &Properties, &Queue);
-
-    // If creating out-of-order queue failed and this property is not
-    // supported (for example, on FPGA), it will return
-    // UR_RESULT_ERROR_INVALID_QUEUE_PROPERTIES and will try to create in-order
-    // queue.
-    if (!MEmulateOOO && Error == UR_RESULT_ERROR_INVALID_QUEUE_PROPERTIES) {
-      MEmulateOOO = true;
-      Queue = createQueue(QueueOrder::Ordered);
-    } else {
-      Adapter->checkUrResult(Error);
-    }
+    Adapter->checkUrResult(Error);
```

Comment:
L541-L543 are equivalent to just `Adapter->call<...>`

Reply:
Good catch. Tweaked this to call the checking version.